src/main/java/mate/jdbc/dao/CarDaoImpl.java

```diff
@@ -0,0 +1,204 @@
+import java.sql.Statement;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import mate.jdbc.exception.DataProcessingException;
+import mate.jdbc.lib.Dao;
+import mate.jdbc.model.Car;
+import mate.jdbc.model.Driver;
+import mate.jdbc.model.Manufacturer;
+import mate.jdbc.util.ConnectionUtil;
+
+@Dao
+public class CarDaoImpl implements CarDao {
+    @Override
+    public Car create(Car car) {
+        String createQuery = "INSERT INTO cars (model,manufacturer_id)"
+                + " VALUES (?,?)";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement preparedStatement =
+                        connection.prepareStatement(createQuery, Statement.RETURN_GENERATED_KEYS)) {
+            preparedStatement.setString(1, car.getModel());
+            preparedStatement.setObject(2, car.getManufacturer().getId());
+            preparedStatement.executeUpdate();
+            ResultSet resultSet = preparedStatement.getGeneratedKeys();
+            while (resultSet.next()) {
+                car.setId(resultSet.getObject(1, Long.class));
+            }
+            addDriversToCar(car);
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can`t create car: " + car, e);
+        }
+        return car;
+    }
+
+    @Override
+    public Optional<Car> get(Long id) {
+        String query = "SELECT * FROM cars AS c "
+                + "JOIN manufacturers AS m ON c.manufacturer_id = m.id "
+                + "WHERE c.id = ? AND c.is_deleted = false;";
+        Car car = null;
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement preparedStatement = connection.prepareStatement(query)) {
+            preparedStatement.setObject(1, id);
+            ResultSet resultSet = preparedStatement.executeQuery();
+            while (resultSet.next()) {
+                car = getCarFromResultSet(resultSet);
+            }
+        } catch (SQLException throwables) {
+            throw new DataProcessingException("Can`t get car by id:" + id, throwables);
+        }
+        car.setDrivers(getDriversForCar(id));
+        return Optional.ofNullable(car);
+    }
+
+    @Override
+    public List<Car> getAll() {
+        String query = "SELECT * "
+                + "FROM cars AS c "
+                + "JOIN manufacturers AS m "
+                + "ON c.manufacturer_id = m.id "
+                + "WHERE c.is_deleted = false;";
+        List<Car> cars = new ArrayList<>();
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement preparedStatement = connection.prepareStatement(query)) {
+            ResultSet resultSet = preparedStatement.executeQuery();
+            while (resultSet.next()) {
+                Car car = getCarFromResultSet(resultSet);
+                cars.add(car);
+            }
+        } catch (SQLException throwables) {
```

Comment:
```suggestion
        } catch (SQLException e) {
```
the same for other methods